Wikipedia/Code/ArticleViewController+Editing.swift

```diff
@@ -80,49 +80,60 @@ extension ArticleViewController {
     
     private func presentEditor(editorViewController: UIViewController) {
         let presentEditorAction = { [weak self] in
-              guard let self else { return }
+            guard let self else { return }
             let navigationController = WMFComponentNavigationController(rootViewController: editorViewController, modalPresentationStyle: .overFullScreen)
-            
+
             let needsIntro = !UserDefaults.standard.didShowEditingOnboarding
             if needsIntro {
                 let editingWelcomeViewController = EditingWelcomeViewController(theme: self.theme) {
-                    
+
                     self.present(navigationController, animated: true)
                 }
                 editingWelcomeViewController.apply(theme: self.theme)
                 self.present(editingWelcomeViewController, animated: true) {
                     UserDefaults.standard.didShowEditingOnboarding = true
                 }
-                
+
             } else {
                 self.present(navigationController, animated: true)
             }
         }
-        
+
         guard let navigationController else { return }
-        
-        if !authManager.authStateIsPermanent {
-            let tempAccountsCoordinator = TempAccountSheetCoordinator(
-                navigationController: navigationController,
-                theme: theme,
-                dataStore: dataStore,
-                didTapDone: { [weak self] in
-                    self?.dismiss(animated: true)
-                },
-                didTapContinue: { [weak self] in
-                    self?.dismiss(animated: true, completion: {
-                        presentEditorAction()
-                    })
-                },
-                isTempAccount: authManager.authStateIsTemporary
-            )
-            
-            _ = tempAccountsCoordinator.start()
-        } else {
-            presentEditorAction()
+
+        state = .loading
+
+        Task {
+            let dataController = WMFTempAccountDataController.shared
+            dataController.checkWikiTempAccountAvailability(language: articleLanguageCode, isCheckingPrimaryWiki: false)
```

Comment:
I'm still having trouble here. There's no completion block or await, how can you be sure the network call has completed before checking the `wikisWithTempAccountsEnabled ` property?

Try this:
- Ensure app is logged out. Add Test as a second language. Re-run from Xcode.
- Edit an EN Wiki article. Temp accounts sheet should not appear.
- Edit a Test Wiki article. Temp accounts sheet does not appear for me, but it should. This is because the first time the flag is not saved in-memory yet and you are checking property before the network call has finished.
- Edit a Test Wiki article again, temp accounts sheet appears.